### DISCUSSION TITLE
What can cause PrjWriteFile() to return ERROR_INVALID_HANDLE in subsequent writes?

### DISCUSSION THREAD

**Discussion by ZabirHoque** [microsoft/ProjFS-Managed-API#88] [Q&A]
During a file recall of particularly large files > 2GB, after a number of successful calls to PrjWriteFile() I receive ERROR_INVALID_HANDLE. Any ideas what can cause this?

**Comment by cgallred** [comment]
As long as your provider hasn't completed the GetFileData callback, the file object that PrjWriteFile uses should remain good. Is it possible your GetFileData callback is calling PrjCompleteCommand before all the data is written out?

**Reply by ZabirHoque** [reply]
We are not calling PrjCompleteCommand() early that I can tell from the log time stamps. Main thing we found that causes this repro is if the user "messes" with the pending file open - specifically, if the file is a .zip and is being opened by Windows file explorer and during the transfer, the user attempts to interact with or drag around the shell zip file explorer and once that windows hangs, use task manager to kill explorer.exe

Is there any etw event stream I can turn on to maybe get some more insight to what could be happening?
